src/backends/mock_main.rs

```diff
@@ -46,6 +65,75 @@ pub struct MockMainPod {
     statements: Vec<Statement>,
 }
 
+impl fmt::Display for MockMainPod {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        writeln!(f, "MockMainPod ({}):", self.id)?;
+        // TODO print input signed pods id and type
+        // TODO print input main pods id and type
+        let offset_input_main_pods = self.offset_input_main_pods();
+        let offset_input_statements = self.offset_input_statements();
+        let offset_public_statements = self.offset_public_statements();
+        for (i, st) in self.statements.iter().enumerate() {
+            if (i < self.offset_input_main_pods()) && (i % self.params.max_signed_pod_values == 0) {
+                writeln!(
+                    f,
+                    "  from input SignedPod {}:",
+                    i / self.params.max_signed_pod_values
+                )?;
+            }
+            if (i >= offset_input_main_pods)
+                && (i < offset_input_statements)
+                && (i % self.params.max_public_statements == 0)
+            {
+                writeln!(
+                    f,
+                    "  from input MainPod {}:",
+                    (i - offset_input_main_pods) / self.params.max_signed_pod_values
+                )?;
+            }
+            if i == offset_input_statements {
+                writeln!(f, "  private statements:")?;
+            }
+            if i == offset_public_statements {
+                writeln!(f, "  public statements:")?;
+            }
+
+            let op = (i >= offset_input_statements)
+                .then(|| &self.operations[i - offset_input_statements]);
+            if !(!f.alternate() && st.is_none()) {
```

Comment:
Nitpick: The check `!(!f.alternate() && st.is_none())` is also done in `fmt_statement_index`

Reply:
right, it was inherited from the [previous version](https://github.com/0xPARC/pod2/blob/f23a78dcb824b21a978b65e738aaca27f8ac9122/src/backends/mock_main.rs#L404), updated it now [at this amend](https://github.com/0xPARC/pod2/compare/40e7051c9f5cf61cbf4e3d2f264c08d4b425f666..36aa5f0d9f054a4f84882b77a23637dffa7cb6f9)